includes/cache/GenderCache.php

```diff
@@ -110,26 +122,32 @@ public function doQuery( $users, $caller = '' ) {
 			}
 		}
 
-		if ( count( $users ) === 0 ) {
-			return;
-		}
+		return $users;
+	}
 
-		$dbr = wfGetDB( DB_SLAVE );
-		$table = array( 'user', 'user_properties' );
-		$fields = array( 'user_name', 'up_value' );
-		$conds = array( 'user_name' => $users );
-		$joins = array( 'user_properties' =>
-			array( 'LEFT JOIN', array( 'user_id = up_user', 'up_property' => 'gender' ) ) );
+	/**
+	 * @param $userNames
+	 * @param $caller
+	 * @return ResultWrapper
+	 */
+	private function getGenderOfUsersFromDB( $userNames, $caller ): ResultWrapper {
+		global $wgExternalSharedDB;
+		$dbr = wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+		$table = [ 'user', 'user_properties' ];
+		$fields = [ 'user_name', 'up_value' ];
+		$conds = [ 'user_name' => $userNames ];
+		$joins = [
+			'user_properties' => [
+				'LEFT JOIN', [ 'user_id = up_user', 'up_property' => 'gender' ],
```

Comment:
I think we should be vary of joining against `wikicities.user` table as we do not know yet what will be the future single source of truth for user info.

Reply:
So what should I use instead? Separated select and match results?

Reply:
Let's keep it for now.  I will file a ticket for refining this after the initial release. https://wikia-inc.atlassian.net/browse/SUS-2976